java-checks/src/main/java/org/sonar/java/checks/AssignmentInSubExpressionCheck.java

```diff
@@ -130,6 +131,11 @@ private static boolean isRelationalExpression(Tree tree) {
   }
 
   @Override
+  public void visitWhileStatement(WhileStatementTree tree) {
```

Comment:
This is way too broad and will results in false negative :

{code}
while(true) {
   if ((a = b = 0) != null) { // Noncompliant but not detected.
 }
{code}